operations/src/test/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkServiceTest.java

```diff
@@ -0,0 +1,231 @@
+    private static final String CHILD1_UUID = "83c2d7f8-2e6b-4f0b-ab7e-7397969c0682";
+    private static final String CHILD2_UUID = "0e33ad0b-7a16-4bfa-b833-6126c262d889";
+    private static final String FILENAME1 = "filename1.txt";
+    private static final String FILENAME2 = "filename2.txt";
+    private AutoCloseable closeable;
+    private PID parentPid;
+    private PID fileObject1Pid;
+    private PID fileObject2Pid;
+    private DownloadBulkService service;
+    private DownloadBulkRequest request;
+    @Mock
+    private AccessControlService aclService;
+    @Mock
+    private RepositoryObjectLoader repoObjLoader;
+    @Mock
+    private AccessGroupSet mockAccessSet;
+    @Mock
+    private AgentPrincipals mockAgent;
+    @Mock
+    private WorkObject parentWork;
+    @Mock
+    private FileObject fileObject1, fileObject2;
+    @TempDir
+    public Path zipStorageBasePath;
+
+    @BeforeEach
+    public void init() throws Exception {
+        closeable = openMocks(this);
+
+        service = new DownloadBulkService();
+        service.setAclService(aclService);
+        service.setRepoObjLoader(repoObjLoader);
+        service.setBasePath(zipStorageBasePath.toString());
+        parentPid = PIDs.get(PARENT_UUID);
+        fileObject1Pid = PIDs.get(CHILD1_UUID);
+        fileObject2Pid = PIDs.get(CHILD2_UUID);
+        request = new DownloadBulkRequest(PARENT_UUID, mockAgent);
+
+        when(mockAgent.getUsername()).thenReturn("user");
+        when(mockAgent.getPrincipals()).thenReturn(mockAccessSet);
+        when(fileObject1.getPid()).thenReturn(fileObject1Pid);
+        when(fileObject2.getPid()).thenReturn(fileObject2Pid);
+    }
+
+    @AfterEach
+    public void after() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void noAccessToWorkObjectTest() {
+        Assertions.assertThrows(AccessRestrictionException.class, () -> {
+            doThrow(new AccessRestrictionException()).when(aclService).assertHasAccess(
+                    anyString(), eq(parentPid), any(), eq(Permission.viewOriginal));
+
+            service.downloadBulk(request);
+        });
+    }
+
+    @Test
+    public void notAWorkTest() {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(null);
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            service.downloadBulk(request);
+        });
+    }
+
+    @Test
+    public void workObjectDoesNotHaveFileObjectsTest() throws IOException {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(parentWork);
+        service.downloadBulk(request);
+        // the zip file should be empty
+        assertZipFile(0, List.of());
+    }
+
+    @Test
+    public void noAccessToFileObjectsTest() throws IOException {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1));
```

Comment:
we should also construct binary so it's clear the access control filter is blocking adding the file rather than the fact there isn't a binary associated with the FileObject.